workspace/lib/xod/core/continuously-pausable/any.cpp

```diff
@@ -0,0 +1,15 @@
+struct State {
+};
+
+{{ GENERATED_CODE }}
+
+void evaluate(Context ctx) {
+  bool isEnabled = getValue<input_EN>(ctx);
+
+  if (isInputDirty<input_EN>(ctx) && !isEnabled) {
+    clearTimeout(ctx);
+  } else if (isTimedOut(ctx) && isEnabled) {
+    emitValue<output_TICK>(ctx, 1);
+    setTimeout(ctx, 0);
+  }
```

Comment:
Not sure, but isn’t a simpler (?) alternative is equivalent?

```cpp
bool isEnabled = getValue<input_EN>(ctx);
if (isEnabled) {
    if (isTimedOut(ctx))
        emitValue<output_TICK>(ctx, 1);
    setTimeout(ctx, 0);
} else {
    clearTimeout(ctx);
}
```

Eliminates dirtiness check.

Reply:
Yes!
I think we can even drop `isTimedOut` check and emit a pulse at the very moment EN is set to true.

Reply:
Indeed! :+1: